site/content/extend/plugins/developer-workflow.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "Developer Workflow"
+date: 2020-07-11T23:00:00-04:00
+weight: 3
+subsection: Plugins
+---
+
+### Common `make` commands
+
+- `make dist` - Compile the plugin into a g-zipped file, ready to upload to a Mattermost server. The file is saved in the plugin repo's `dist` folder.
+- `make deploy` - Compiles the plugin using the `make dist` command, then automatically deploys the plugin to the Mattermost server
+- `make watch` - Uses webpack's watch feature to re-compile and deploy the webapp portion of your plugin on any change to the `webapp/src` folder
+- `make test` - Runs the plugin's server tests and webapp tests
+- `make check-style` - Runs linting checks on the plugin's server and webapp folders
+- `make enable` - Enables the plugin on the Mattermost server
+- `make disable` - Disables the plugin on the Mattermost server
+- `make reset` - Disables and re-enables the plugin on the Mattermost server
+- `make debug-plugin` - Starts a `delve` process and attaches it to your running plugin
+
+You can run the development build of the plugin by setting the environment variable `MM_DEBUG=1`, or prefixing the variable at the beginning of the `make` command. For example, `MM_DEBUG=1 make deploy` will deploy the development build of the plugin to your server, allowing you to have a more fluid debugging experience. To instead use the production build of the plugin, unset the `MM_DEBUG` environment variable before running the `make` commands.
+
+### Exposing the Mattermost server using `ngrok`
+
+When a public integrates with an external service, webhooks and/or authentication redirects are necessary, which requires your local server to be available on the web. In order for your Mattermost server to be available to process webhook requests, it needs to expose its port to an external address. A common way to do this is to use the command line tool [ngrok](https://ngrok.com). Follow these steps to set up `ngrok` with your server:
```

Comment:
I think there's a missing word after "public".

Reply:
That's supposed to say "plugin" instead of "public" :)